### PR TITLE
zsh-powerlevel10k init at 2019-07-28

### PR DESCRIPTION
**Pull request by baka-yuki**
###### Motivation for this change
zsh-powerlevel9k can increase latency with large git repositories. Powerlevel10k is a drop in replacement that decreases that latency.

Additionally, I have added myself to the maintainers file. I am planning on (fingers crossed) submitting more than just this pull, so I hope this is appropriate. :sweat_smile: 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by zyradyl** [comment]
I have updated the version being used per #65431, I'll update every seven days till merger. :smile:

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5859,6 +5859,15 @@
     github = "zraexy";
     name = "David Mell";
   };
+  zyradyl = {
```
Comment (ryantm) [inline maintainers/maintainer-list.nix]: Please separate the maintainer addition into a separate commit.

File: pkgs/shells/zsh/zsh-powerlevel10k/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# To make use of this derivation, use
+# `programs.zsh.promptInit = "source ${pkgs.zsh-powerlevel10k}/share/zsh-powerlevel10k/powerlevel10k.zsh-theme";`
+
+let
+  version = "2019-07-28";
+in
+
+stdenv.mkDerivation rec {
+  name = "powerlevel10k-${version}";
```
Comment (ryantm) [inline pkgs/shells/zsh/zsh-powerlevel10k/default.nix]:
```suggestion
  pname = "powerlevel10k";
  version = "2019-07-28";
```

File: pkgs/shells/zsh/zsh-powerlevel10k/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  src = fetchFromGitHub {
+    owner = "romkatv";
+    repo = "powerlevel10k";
```
Comment (ryantm) [inline pkgs/shells/zsh/zsh-powerlevel10k/default.nix]:
```suggestion
    repo = pname;
```

**Comment by ryantm** [comment]
@GrahamcOfBorg build zsh-powerlevel10k

**Review by ryantm — CHANGES_REQUESTED** [review]
It doesn't seem to build.

**Comment by flokli** [comment]
I assume this is superseded by #76386.
